website/src/components/Stats/Stats.components.tsx

```diff
@@ -38,30 +39,208 @@ type MessageTreeStateStatsProps = {
   };
 };
 
+const findLangRow = (rows: string[][], lang: string): number => {
+  for (let i = 0; i < rows.length; i++) {
+    const row_lang = rows[i][0];
+    if (row_lang === lang) return i;
+  }
+
+  return -1;
+};
+
+function getTableData(
+  stat: StatType & {
+    stats: {
+      lang: string;
+      state: string;
+      count: number;
+    }[];
+  },
+  t
+) {
+  // The headers come from the message tree State.
+  // These two states are not considered because they make
+  // the table too wide and they are not so interesting:
+  // - "ready_for_scoring",
+  // - "scoring_failed"
+  let headers = [
+    "language", // This one doesn't come from the message tree state
+    "prompt_lottery_waiting",
+    "initial_prompt_review",
+    "growing",
+    "backlog_ranking",
+    "ranking",
+    "ready_for_export",
+    "aborted_low_grade",
+    "halted_by_moderator",
+  ];
+
+  const languages = [];
+  for (let i = 0; i < stat.stats.length; i++) {
+    if (!languages.includes(stat.stats[i].lang)) {
+      languages.push(stat.stats[i].lang);
+    }
+  }
```

Comment:
this is trying to get unique languages?
```js
const languages = [... new Set(stat.stats.map(s => s.lang))]
```

Reply:
I was looking for someway of using set, as I would do in Python. I tried your approach, but I got an error about needing option `--downlevelIteration` for using it with a `Set`. I used `Array.from` instead.